oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/NodeDocumentSweeper2.java

```diff
@@ -0,0 +1,357 @@
+/*
```

Comment:
Agree, and I was aware of this. However, I did this on purpose as indicated in the class comment. The alternative would have been to reuse NodeDocumentSweeper - with the risk or cost that any future change on that class would have to be carefully crafted to avoid any side-effects on NodeDocumentSweeper2 behaviour. I considered this future risk as higher versus the code-duplication. Wdyt?

Reply:
Agreed.